src/Auth/HttpBasicAuth.php

```diff
@@ -0,0 +1,108 @@
+<?php
+namespace Yiisoft\Yii\Web\Auth;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Yiisoft\Yii\Web\User\IdentityInterface;
+use Yiisoft\Yii\Web\User\IdentityRepositoryInterface;
+
+/**
+ * HttpBasicAuth is an action filter that supports the HTTP Basic authentication method.
+ *
+ * > Tip: In case authentication does not work like expected, make sure your web server passes
+ * username and password to `$_SERVER['PHP_AUTH_USER']` and `$_SERVER['PHP_AUTH_PW']` variables.
+ * If you are using Apache with PHP-CGI, you might need to add this line to your `.htaccess` file:
+ * ```
+ * RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization},L]
+ * ```
+ */
+class HttpBasicAuth implements AuthInterface
```

Comment:
@samdark should auth methods be `final` as well?

Reply:
Let's make them final initially. It is possible to change on request.